main/lsp/LocalVarFinder.h

```diff
@@ -27,7 +27,7 @@ class LocalVarFinder {
     ast::ExpressionPtr preTransformClassDef(core::Context ctx, ast::ExpressionPtr classDef);
     ast::ExpressionPtr postTransformClassDef(core::Context ctx, ast::ExpressionPtr classDef);
 
-    const std::vector<core::LocalVariable> &result() const;
+    const std::vector<core::NameRef> &result() const;
```

Comment:
How are you planning to change this to account for instance variables? I hope that it'll still be possible to have separate results for e.g. "all local variables" and "all instance variables", not one big `vector<NameRef>` that mixes the two.

Reply:
I was going to have a separate `FieldFinder`, since instance/class variables are not limited to a target method, but to an entire class, and globals are, well, globals.  (I am not 100% sure how to handle globals yet, but I am less concerned about them.)  There's similar logic, but I don't think there's enough to stuff them into a single class.

AFAICT, it will still be possible to have separate "all local variables" and "all instance variables", but you can't complete both sets in the same list of completion results, because instance variables are effectively namespaced by virtue of their leading `@`.  (If `completion.cc` implemented a fuzzy completion algorithm, rather than the prefix-based algorithm it has, then you could of course complete from both sets in the same list.)  So the methods for finding them are slightly different, but as far as completion code is concerned, they can be treated (almost) identically after the lists of potential completions are found.

(This would also be an explanation as to why I'm not renaming it `VariableFinder`, since `LocalVarFinder` is still only for local variables; I would consider `LocalNameFinder` as a reasonable substitute.)